Dockerfile

```
@@ -0,0 +1,42 @@
+FROM python:3.11.9-slim-bullseye as base
+RUN apt-get update && apt-get install curl -y
+RUN export POETRY_HOME=/usr/local; curl -sSL https://install.python-poetry.org | python3.11 -
+EXPOSE 8000
+WORKDIR /opt/todoapp
+COPY ["poetry.lock", "poetry.toml", "pyproject.toml", "./"]
+RUN poetry install --no-dev --no-root
+
+FROM base as development
+ENV FLASK_DEBUG="true"
+RUN poetry install
+Entrypoint poetry run flask run --host=0.0.0.0
+
+FROM base as debug
+ENV FLASK_DEBUG="true"
+RUN poetry install
+Entrypoint tail -f /dev/null
+
+FROM base as production
+RUN poetry install --no-dev
+COPY ./todo_app ./todo_app
+ENTRYPOINT poetry run gunicorn --bind 0.0.0.0 'todo_app:app'
```

This crashed when I tried to run it using docker compose; the fix I chose was to switch it over to load the app directly but I'll admit I haven't looked deeper given the setup seemed to work OK via Ansible
```suggestion
ENTRYPOINT poetry run gunicorn --bind 0.0.0.0 'todo_app.app:create_app()'
```